### PR TITLE
Moves dependencies to their own gradle file, and adds capabilities to link to ntcore and wpilib

### PR DESCRIPTION
**Pull request by ThadHouse**

### CODE REVIEW

File: dependencies.gradle [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+            linker.args "${openCvNativesFolder}/libopencv_videoio.a"
+            linker.args "${openCvNativesFolder}/libopencv_videostab.a"
+            linker.args "${openCvNativesFolder}/libopencv_core.a"
+            linker.args "${openCvNativesFolder}/liblibjpeg.a"
+            linker.args "${openCvNativesFolder}/liblibpng.a"
+            linker.args "-ldl"
+            linker.args "-lz"
+        }
+    }
+}
```
Comment (PeterJohnson) [inline dependencies.gradle]: EOL at EOF

Comment (ThadHouse) [reply dependencies.gradle]: Fixed

File: build.gradle [hunk 2]
```diff
@@ -50,243 +50,7 @@ ext.getPlatformPath = { binary ->
...
-            linker.args "${openCvNativesFolder}/liblibpng.a"
-            linker.args "-ldl"
-            linker.args "-lz"
-        }
-    }
-}
+apply from: "dependencies.gradle"
 
 ext.setupDefines = { project, binaries ->
     binaries.all {
```
Comment (PeterJohnson) [inline build.gradle:56]: The addWpiUtilLibraryLinks on line 63 needs to be updated to use shared libs on arm (e.g. `if (targetPlatform.architecture.arm)` and static libs on other platforms.

Comment (ThadHouse) [reply build.gradle:56]: It looks like you actually double add the links. the projects in cameraserver.gradle add the links, and so does setup defines. Which ones should we keep, as we don't need both.

Also I think I'd prefer to do the same trick I did in ntcore, where only specifically the roboRio builds use shared. That way devices like coprocessors can use static.

Comment (ThadHouse) [reply build.gradle:56]: I decided to go with the setup defines link, and used the trick to make it so only the roboRio builds use a shared wpiutil.

**Review by PeterJohnson — APPROVED** [review]

**Pull request merged**